### PR TITLE
1583 Frontend tour

### LINKED ISSUE #1583: Improve UI/UX (esp. for new users)
- [x] I have searched the [issues](https://github.com/openkfw/TruBudget/issues) of this repository and believe that this is not a duplicate.

## Description 💡

List of ideas how to improve user experience/onboarding.

- [ ] add tooltips to (interactive) elements
- [ ] guided tour using intro.js, joyride, similar (minus side: has to be maintained)
- [ ] ...

### PR DESCRIPTION
**Pull request by galethil**
### Checklist

- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/openkfw/TruBudget/blob/main/.github/CONTRIBUTING.md#open-a-pull-request).
- [ ] I fixed all necessary PR warnings
- [ ] The commit history is clean
- [ ] The E2E tests are passing
- [ ] If possible, the issue has been divided into more subtasks
- [ ] I did a self review before requesting a review from another team member

### How to test

1. Open main page
2. Open main menu
3. Click on "Start tour"
4. Follow instructions
5. Try different start points, go there and back, try to break it :)

Closes #1583

### CODE REVIEW

**Comment by MartinJurcoGlina** [comment]
any plans adding translations?

File: frontend/src/pages/Main/TourWrapper.scss [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  background-color: #fff;
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
+  padding: 10px;
+  font-size: 14px;
+  line-height: 1.5;
+  color: #333;
+  min-width: 400px;
+  min-width: 200px;
```
Comment (pmolnar-dev) [inline frontend/src/pages/Main/TourWrapper.scss:12]: which of these min-width should be used?

File: frontend/src/pages/Main/TourWrapper.js [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+  }
+
+  // If no object with "url" attribute is found, return null
+  return null;
+}
+
+const initialState = ({ firstProjectId, firstSubprojectId }) => {
+  const beforeStart = [
+    {
+      navigateTo: "/projects"
```
Comment (MartinJurcoGlina) [inline frontend/src/pages/Main/TourWrapper.js:140]: /projects could be constant as it appears several times below

File: frontend/src/pages/Main/TourWrapper.js [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+    }
+  ];
+  const steps = [
+    {
+      target: "body",
+      content: "Welcome to TruBudget application tour.",
```
Comment (MartinJurcoGlina) [inline frontend/src/pages/Main/TourWrapper.js:146]: all the content texts should come from translations

File: frontend/src/pages/SubProjects/ProjectDetails.js [hunk 4]
```diff
@@ -77,6 +78,25 @@ const ProjectDetails = (props) => {
...
+    state: { tourActive },
+    goToNextStepIf
+  } = useTourAppContext();
+
+  const ref = useRef();
+
+  useEffect(() => {
+    if (tourActive && !ref.current) {
+      goToNextStepIf();
+      setTimeout(() => {
```
Comment (MartinJurcoGlina) [inline frontend/src/pages/SubProjects/ProjectDetails.js:93]: timeout should be cleared in useEffect return

File: frontend/src/pages/Workflows/SubProjectDetails.js [hunk 5]
```diff
@@ -74,6 +82,25 @@ const SubProjectDetails = ({
...
+    state: { tourActive },
+    goToNextStepIf
+  } = useTourAppContext();
+
+  const ref = useRef();
+
+  useEffect(() => {
+    if (tourActive && !ref.current) {
+      goToNextStepIf();
+      setTimeout(() => {
```
Comment (MartinJurcoGlina) [inline frontend/src/pages/Workflows/SubProjectDetails.js:97]: clear timeout in return

**Review by MartinJurcoGlina — CHANGES_REQUESTED** [review]

**Review by MartinJurcoGlina — APPROVED** [review]

**Pull request merged**